src/components/App/index.js

```diff
@@ -1,39 +1,76 @@
-import React, {Component} from 'react';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import { today } from 'constants/apiQueries';
+import { bindActionCreators } from 'redux';
 
+import { today } from 'constants/apiQueries';
 import { fetchEpisodes } from 'redux/actions/episodesActions';
 
+import ActiveEpisodeModal from 'components/App/ActiveEpisodeModal';
 import EpisodesList from 'components/App/EpisodesList';
 import Header from 'components/App/Header';
+import Pagination from 'components/App/Pagination';
 import Spinner from 'components/App/Spinner';
 
 import styles from './App.module.scss';
 
 class App extends Component {
+  state = {
+    isModalVisible: false,
+    activeEpisode: null,
+  }
+
   componentDidMount() {
-    this.props.dispatch(fetchEpisodes());
+    this.props.fetchEpisodes();
+  }
+
+  openModal = (episode) => {
+    this.setState({ 
+      isModalVisible: true,
+      activeEpisode: episode
+    });
+  }
+
+  closeModal = event => {
+    const { id } = event.target;
+    if (id === "modalBg" || id === "closeModal") this.setState({ isModalVisible: false })
   }
 
   render() {
-    const { error, loading, episodes } = this.props;
+    const { error, loading, episodes, currentPage, episodesPerPage } = this.props;
+    const {isModalVisible, activeEpisode } = this.state;
+    const indexOfLastEpisode = parseInt(currentPage) * episodesPerPage;
```

Comment:
logic for pagination maybe can be extract somewhere? consider use `reselect` lib

Reply:
Good idea. :) I used it, for the first time and hope it's fine the way I handled it. I needed currentEpisodes on the top level component together with other state elements but I assigned it a result of selector's computing. Please let me know what do you think about it.